### PR TITLE
Improvements to the NOTICE default FreeMarker template

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: reporter/src/main/resources/templates/notice/default.ftl [hunk 1]
```diff
@@ -62,10 +62,12 @@ ${exceptionText}
         [/#if]
     [/#if]
 [/#list]
-[#-- Add the licenses of all dependencies. --]
+[#-- Add the licenses of all packages. --]
 [#if packages?has_content]
+[#if hasNoticeProjectLicenses]
```
Comment (MarcelBochtler) [inline reporter/src/main/resources/templates/notice/default.ftl:64]: I think the indentation is wrong for this element?!

Comment (sschuberth) [reply reporter/src/main/resources/templates/notice/default.ftl]: Hah, good eyes, fixed!

File: reporter/src/main/resources/templates/notice/default.ftl [hunk 2]
```diff
@@ -67,14 +67,18 @@ ${exceptionText}
     [/#if]
 This software depends on external packages and source code.
 The applicable license information is listed below:
-[/#if]
 
+[/#if]
```
Comment (MarcelBochtler) [inline reporter/src/main/resources/templates/notice/default.ftl:71]: is this change intentional?

Comment (sschuberth) [reply reporter/src/main/resources/templates/notice/default.ftl:71]: Yes, IIRC this new blank line inside the `[/#if]` is a substitute for the blank line in former line 69 outside the `[/#if]`. TBH, I also only found out by running the tests and tweaking the output.

**Review by MarcelBochtler — APPROVED** [review]
